### DISCUSSION TITLE
Is that possible to use own segmentation for diarization?

### DISCUSSION THREAD

**Discussion by Bodawen** [pyannote/pyannote-audio#1547] [Q&A] [answered]
Thanks for the project.

I'd like to ask if it is possible to use my segmentation data instead of the segmentation model in the diarization pipeline?

**Comment by hbredin** [comment]
I guess you can try and plug your data [here](https://github.com/pyannote/pyannote-audio/blob/6ad2f87015d6cfe660a098233ff2f72bbdc5cd84/pyannote/audio/pipelines/speaker_diarization.py#L200-L225).

**Reply by Bodawen** [reply]
@hbredin Thanks for the reply. I checked the code, but it is still not clear how to convert human-readable data such as .rttm to the **SlidingWindowFeature** object. Can you give me more information about that?

**Reply by hbredin** [reply, marked as answer]
No, sorry. I don't have much time to spend on this.
But, [this blog post](https://herve.niderb.fr/fastpages/2022/10/23/One-speaker-segmentation-model-to-rule-them-all) might help.

**Reply by Bodawen** [reply]
Thanks I will read it.
